src/hopen/renderer/xf.cljc

```diff
@@ -127,6 +131,11 @@
 (defn- inline-quote [env val]
   val)
 
+(defn- ctx-lookup
+  "Tries to look up a field name in every context starting from most nested context first."
+  [env field-name]
+  (some #(get % (keyword field-name)) (reverse (get-in env [:bindings 'hopen/ctx] []))))
```

Comment:
This function may only be used by the Mustache syntax. I would prefer it to live in one of the mustache syntax's namespace and not been added to the default-env of this namepace.

It means that you would need to define a default-env in the mustache syntax's namespace as well, which you could expression as an update of the one which is here.

Reply:
update: the code being refactored, I mean based on `hopen.renderer.env`.